app/models/report.rb

```diff
@@ -49,7 +49,7 @@ class Report < ActiveRecord::Base
   scope :completed, -> { where(status: %w(completed archived)) }
 
   scope :pending, lambda {
-    joins(:dispatches).where(dispatches: {status: 'pending'})
+    joins(:dispatches).where(dispatches: {status: 'pending'}).distinct
```

Comment:
LGTM

El El mar, ago 23, 2016 a las 4:13 PM, Hy Carrel <notifications@github.com>
escribió:

> In app/models/report.rb
> <https://github.com/egavas629/concrn-rails/pull/54#discussion_r75969037>:
>
> > @@ -49,7 +49,7 @@ class Report < ActiveRecord::Base
> >    scope :completed, -> { where(status: %w(completed archived)) }
> >
> >    scope :pending, lambda {
> > -    joins(:dispatches).where(dispatches: {status: 'pending'})
> > +    joins(:dispatches).where(dispatches: {status: 'pending'}).distinct
>
> The way we track the status of multiple responders for a single report is
> by a record in the dispatches table for the status of each responders for a
> given report matched the report as a foreign key. We want to return a
> single copy of each report, with the dispatches records joined to it. We
> think distinct provides the correct behavior.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/egavas629/concrn-rails/pull/54/files/6ad24041fdc3e11335df6a930caa6e34f8b24611#r75969037>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAxb4Xe1QGrkf80yEBsCPAKDjd2iZda9ks5qi36WgaJpZM4JnHEl>
> .
>